account-kit/core/src/actions/createAccount.ts

```diff
@@ -94,18 +97,26 @@ export async function createAccount<TAccount extends SupportedAccountTypes>(
     throw new Error("Signer not connected");
   }
 
-  const cachedAccount = accounts[chain.id]?.[type];
+  const cachedAccount = (
+    type === "ModularAccountV2"
+      ? accounts[chain.id]?.["ModularAccountV2"]?.[mode]
+      : accounts[chain.id]?.[type]
+  ) as AccountState<TAccount>;
+
   if (cachedAccount.status !== "RECONNECTING" && cachedAccount.account) {
     return cachedAccount.account;
   }
-  const cachedConfig = store.getState().accountConfigs[chain.id]?.[type];
+  const cachedConfig =
+    type === "ModularAccountV2"
+      ? store.getState().accountConfigs[chain.id]?.["ModularAccountV2"]?.[mode]
+      : store.getState().accountConfigs[chain.id]?.[type];
 
   const accountPromise = (() => {
     switch (type) {
       case "LightAccount":
         return createLightAccount({
           ...params,
-          ...cachedConfig,
+          ...(cachedConfig as OmitSignerTransportChain<CreateLightAccountParams>),
```

Comment:
Same here. Do we want a default config that we fall back too in the case it's undefined?

Reply:
this is the same thing we do for all of the other accounts. i'm not sure why it wasn't done for LightAccount previously. these are optional params.

Reply:
I am curious of that too. I'm also curious why we already have been doing it in the past for all of the other account types here, just not LightAccount. This is the error I get... 🤔 
```ts
Types of property 'version' are incompatible.
  Type '"v1.0.1" | "v1.0.2" | "v1.1.0" | "v2.0.0" | undefined' is not assignable to type '"v2.0.0" | undefined'.
    Type '"v1.0.1"' is not assignable to type '"v2.0.0"'.
```

The error goes away if I cast either the `params` or the `cachedConfig`. (We cast both for all of the other account types already.)

Also, somewhat random, but I believe the type casts for all of the other account types here are reversed, i.e. `params` should be the CreateAccountParams, then `cachedConfig` should be AccountConfig.